### PR TITLE
Add Abyss rifts plugin

### PR DESCRIPTION
**Pull request by Sethtroll**
abyssrifts

### CODE REVIEW

**Comment by Adam-** [comment]
I would like to see a way to write plugins like this without having to do full scene scanning per frame. I think there is a game object spawn event that could be used instead?

**Comment by SomeoneWithAnInternetConnection** [comment]
There's the `GameObejctsChanged` event, but as far as I can tell, that event is only fired for `GameObject`s, and it looks like this plugin only cares about `DecorativeObject`s

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 1]
```diff
@@ -46,12 +57,18 @@
...
+	private final Set<DecorativeObject> abyssObjects = new HashSet<>();
+
 	@Inject
 	RunecraftOverlay overlay;
 
 	@Inject
 	BindNeckOverlay bindNeckOverlay;
 
+	@Inject
+	AbyssOverlay abyssOverlay;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Make the new injected field private (and others in plugin as well)

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssRifts.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	FIRE_RIFT(24971, FIRE_RUNE),
+	LAW_RIFT(25034, LAW_RUNE),
+	MIND_RIFT(25379, MIND_RUNE),
+	NATURE_RIFT(24975, NATURE_RUNE),
+	SOUL_RIFT(25377, SOUL_RUNE),
+	WATER_RIFT(25376, WATER_RUNE);
+
+	@Getter
+	private final int objectId;
+	@Getter
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssRifts.java]: Empty line above annotations

**Pull request merged**